internal/database/repository/queue.go

```diff
@@ -82,7 +90,18 @@ func (s *PostgresRepository) UpsertQueues(ctx context.Context, queues []*dao.Par
 				"created_at":               time.Now().Unix(),
 			})
 		if err != nil {
-			return fmt.Errorf("could not insert/update queue into DB: %v", err)
+			return fmt.Errorf("could not insert/update queue %s into DB: %v", q.QueueName, err)
+		}
+
+		if len(q.Children) > 0 {
+			children := []*dao.PartitionQueueDAOInfo{}
+			for _, child := range q.Children {
+				children = append(children, &child)
+			}
+			err = s.UpsertQueues(ctx, &queueId, children)
+			if err != nil {
+				return fmt.Errorf("could not insert/update one or more children of queue %s into DB: %v", q.QueueName, err)
+			}
```

Comment:
This seems bit tricky. What happens when we do an update? I guess it will pass a newly generated ID from line no 66 to the child, which is incorrect.

Probably we need to use the `RETURNING` keyword in Postgres. Something like,
`INSERT INTO items (name) values ('bear') RETURNING id;`

Reply:
I agree - fixed in commit af53844 .